sycl/source/detail/graph_impl.cpp

```diff
@@ -356,6 +392,19 @@ void graph_impl::makeEdge(std::shared_ptr<node_impl> Src,
   removeRoot(Dest); // remove receiver from root node list
 }
 
+void graph_impl::duplicateGraph(const graph_impl &GraphImpl) {
+  std::map<node_impl *, std::shared_ptr<node_impl>> NodesMaps;
+  for (auto &Root : GraphImpl.MRoots) {
+    auto CpyRoot = Root->duplicateNodeAndSuccessors(NodesMaps);
+    this->addRoot(CpyRoot);
```

Comment:
Another nit but I don't think this kind of abbreviation is worth it, since IMO it hurts readability and only saves a single character and I don't think `CopyRoot` is an unreasonably long name.

Additional super nit, `RootCopy` reads a little better IMO.